### PR TITLE
metainfo: developer_name is deprecated

### PR DESCRIPTION
**Pull request by cbm755**
See [1].

[1] https://freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-developer

### CODE REVIEW

**Comment by cbm755** [comment]
There is a new validator, that can be run like this:

```
$ flatpak run --command=flatpak-builder-lint org.flatpak.Builder appstream io.github.gnu_octave.doctest.metainfo.xml
io.github.gnu_octave.doctest.metainfo.xml: OK
```

These changes still pass the original linter:
```
$ appstream-util validate io.github.gnu_octave.doctest.metainfo.xml 
io.github.gnu_octave.doctest.metainfo.xml: OK
```

File: io.github.gnu_octave.doctest.metainfo.xml [hunk 1]
```diff
@@ -25,11 +25,12 @@ without any warranty.
...
   </keywords>
-  <translation/>
   <url type="homepage">https://octave.sourceforge.io/doctest</url>
   <url type="bugtracker">https://github.com/gnu-octave/octave-doctest/issues/new</url>
   <metadata_license>FSFAP</metadata_license>
   <project_license>BSD-3-Clause</project_license>
-  <developer_name>Octave Community</developer_name>
+  <developer id="octave.org">
+    <name>Octave Community</name>
+  </developer>
```
Comment (cbm755) [inline io.github.gnu_octave.doctest.metainfo.xml:34]: @siko1056 Do you think we should use `id="octave.org"` or `id="gnu-octave.github.io"`  here?  That is, is it important to distinguish between "Octave Community" and "the GNU Octave Project"?

The [docs](https://freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-developer) say:

>  The element should have a id property, containing a unique ID to identify the respective developer. It is recommended to use a web URL, like gnome.org as ID to achieve a higher chance of uniqueness. 

There is also an example at the bottom of that link.  I haven't found many projects that have adopted the new `<developer>` tag yet so not a lot of "best practice" to follow.

Comment (cbm755) [reply io.github.gnu_octave.doctest.metainfo.xml:34]: CC @pr0m1th3as @NRJank @mmuetzel as well.

Comment (pr0m1th3as) [reply io.github.gnu_octave.doctest.metainfo.xml:34]: Should I apply this change to statistics package as well? What is and where this xml file is used for? Can you elaborate, please?

Comment (mmuetzel) [reply io.github.gnu_octave.doctest.metainfo.xml:34]: Maybe, something that could be asked during the developer meeting next Tuesday?

Comment (cbm755) [reply io.github.gnu_octave.doctest.metainfo.xml:34]: Thanks, and I see its already on the agenda!

@pr0m1th3as: let's wait until we have some rough consensus and then do other packages.

  - I don't think there is a huge time pressure here:  Most I didn't find any software on my computer
    ```
    cd /usr/share/metainfo
    $ grep * -rie "<developer_name>" | wc -l
    50
    $ grep * -rie "<developer>" | wc -l
    3
    ```
  - and of those 3, none of them have the `id=` argument, and all have just personal names under developer.
  - Flathub is linting this as a warning, not an error.
      - TODO: send a patch to Octave itself.
  - TODO: can/should one include more than one `<developer>` tags?
  - As for "What is [it] used for?": various packaging tools, especially GUI tools, use these metainfo files  (Gnome Software, Flathub, Elementary OS, probably others).  Personally, I think of it as moving some of tasks of traditional distro packaging efforts upstream.

Comment (siko1056) [reply io.github.gnu_octave.doctest.metainfo.xml:34]: My 2 cents to the discussion:
- Octave packages does currently not parse `metainfo.xml` files.  Is this file even part of the Octave package specification https://docs.octave.org/latest/Creating-Packages.html if not who what requires it?
- I think `octave.org` is more appropriate, but I do not know the specification of this file.  Is it a FreeDesktop standard?  Maybe we should consult their tool: https://www.freedesktop.org/software/appstream/metainfocreator/#/

Comment (cbm755) [reply io.github.gnu_octave.doctest.metainfo.xml:34]: > Is this file even part of the Octave package specification

No.

> if not who what requires it?

I don't think anything requires it.  But It does make the package installable via Gnome Software and other GUI packaging tools, see "Add-ons" in this screenshot:

image

> Is it a FreeDesktop standard?

Yes it is.  My original message linked to the relevant part of the spec: https://freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-developer

**Pull request merged**